src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -206,32 +252,58 @@ static SyntaxNode TryGetAcceptedNodeOrExtracted(SyntaxNode node, Predicate<Synta
+                var extractedNode = extractNode(node, syntaxFacts, extractParentsOfHeader);
                 return (extractedNode != null && predicate(extractedNode))
                     ? extractedNode
                     : null;
             }
         }
 
+        private static TextSpan GetSpanWithoutAttributes(SyntaxNode node, SyntaxNode root, ISyntaxFactsService syntaxFacts)
+        {
+            // Span without AttributeLists
+            // - No AttributeLists -> original .Span
+            // - Some AttributeLists -> (first non-trivia/comment Token.Span.Begin, original.Span.End)
+            //   - We need to be mindful about comments due to:
+            //      // [Test1]
+            //      //Comment1
+            //      [||]object Property1 { get; set; }
+            //     the comment node being part of the next token's (`object`) leading trivia and not the AttributeList's node.
+            var attributeList = syntaxFacts.GetAttributeLists(node);
+            if (attributeList.Any())
+            {
+                var endOfAttributeLists = attributeList.Last().Span.End;
+                var afterAttributesToken = root.FindTokenOnRightOfPosition(endOfAttributeLists);
+
+                return TextSpan.FromBounds(afterAttributesToken.Span.Start, node.Span.End);
+            }
+
+            return node.Span;
+        }
+
         /// <summary>
         /// <para>
-        /// Extractor function for <see cref="TryGetSelectedNodeAsync(Document, TextSpan, Predicate{SyntaxNode}, Func{SyntaxNode, ISyntaxFactsService, SyntaxNode}, CancellationToken)"/> methods 
-        /// that retrieves expressions from  declarations and assignments. Otherwise returns unchanged <paramref name="node"/>.
+        /// Extractor function for <see cref="TryGetSelectedNodeAsync(Document, TextSpan, Predicate{SyntaxNode}, Func{SyntaxNode, ISyntaxFactsService, bool, SyntaxNode}, CancellationToken)"/> method 
+        /// that retrieves nodes that should also be considered as refactoring targets given <paramref name="node"/> is considered. 
+        /// Can extract both nodes above and under given <paramref name="node"/>.
         /// </para>
         /// <para>
-        /// The rationale is that when user selects e.g. entire local delaration statement [|var a = b;|] it is reasonable
+        /// E.g. extracts expressions from assignment statement (`a = b;` -> `b`) or lambda node for its body.
+        /// </para>
+        /// <para>
+        /// The rationale is that when user selects e.g. entire local declaration statement [|var a = b;|] it is reasonable
         /// to provide refactoring for `b` node. Similarly for other types of refactorings.
         /// </para>
         /// </summary>
-        protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts)
+        protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts, bool extractParentsOfHeader)
```

Comment:
when are extractParentsOfHeader set to false or true?

Reply:
True for caret location, false for selections. 

-> 
```csharp
using System;
class C
{
    public [|int|] a { get; set; }
    //vs 
    public i[||]nt a { get; set; }
}
```

Reply:
Added docs (including rationale) for that so that it's absolutely clear.

Reply:
great.  the docs are appreciated.